applications/ParticleMechanicsApplication/custom_utilities/mpm_boundary_rotation_utility.h

```diff
@@ -185,109 +186,30 @@ class MPMBoundaryRotationUtility: public CoordinateTransformationUtils<TLocalMat
-		}
-	}
-
-	// An extra function to distinguish the application of slip in condition considering penalty imposition
-	void ConditionApplySlipCondition(TLocalMatrixType& rLocalMatrix,
-			TLocalVectorType& rLocalVector,
-			GeometryType& rGeometry) const
-	{
-		// If it is not a penalty condition, do as standard
-		if (!this->IsPenalty(rGeometry))
-		{
-			this->ApplySlipCondition(rLocalMatrix, rLocalVector, rGeometry);
-		}
-		// Otherwise, do the following modification
-		else
-		{
-			const unsigned int LocalSize = rLocalVector.size();
-
-			if (LocalSize > 0)
-			{
-				const unsigned int block_size = this->GetBlockSize();
-				TLocalMatrixType temp_matrix = ZeroMatrix(rLocalMatrix.size1(),rLocalMatrix.size2());
-				for(unsigned int itNode = 0; itNode < rGeometry.PointsNumber(); ++itNode)
-				{
-					if(this->IsSlip(rGeometry[itNode]) )
-					{
-						// We fix the first displacement dof (normal component) for each rotated block
-						unsigned int j = itNode * block_size;
-
-						// Copy all normal value in LHS to the temp_matrix
-                        // [ does nothing for dummy rLocalMatrix (size1() == 0) -- RHS only case ]
-						for (unsigned int i = j; i < rLocalMatrix.size1(); i+= block_size)
-						{
-							temp_matrix(i,j) = rLocalMatrix(i,j);
-							temp_matrix(j,i) = rLocalMatrix(j,i);
-						}
-
-						// Remove all other value in RHS than the normal component
-						for(unsigned int i = j; i < (j + block_size); ++i)
-						{
-							if (i!=j) rLocalVector[i] = 0.0;
-						}
-					}
-				}
-                // All entries in penalty matrix zeroed out except for normal component
-                // [ no effect in case of empty dummy rLocalMatrix ]
-				rLocalMatrix = temp_matrix;
-			}
-		}
-	}
-
-	// An extra function to distinguish the application of slip in condition considering penalty imposition (RHS Version)
-	void ConditionApplySlipCondition(TLocalVectorType& rLocalVector,
-			GeometryType& rGeometry) const
-	{
-        // creates an empty dummy matrix to pass into the 'full' ConditionApplySlipCondition -- this dummy matrix is
-        // ignored, effectively only updating the RHS
-        TLocalMatrixType dummyMatrix;
-        this->ConditionApplySlipCondition(dummyMatrix, rLocalVector, rGeometry);
-	}
-
-	// Checking whether it is normal element or penalty element
-	bool IsPenalty(GeometryType& rGeometry) const
-	{
-		bool is_penalty = false;
-		for(unsigned int itNode = 0; itNode < rGeometry.PointsNumber(); ++itNode)
-		{
-			if(this->IsSlip(rGeometry[itNode]) )
-			{
-				const double identifier = rGeometry[itNode].FastGetSolutionStepValue(mrFlagVariable);
-				const double tolerance  = 1.e-6;
-				if (identifier > 1.00 + tolerance)
-				{
-					is_penalty = true;
-					break;
-				}
-			}
-		}
-
-		return is_penalty;
-	}
+    bool IsParticleBased(NodeType& rNode) const
+    {
+        if(this->IsSlip(rNode) )
+        {
+            const double identifier = rNode.FastGetSolutionStepValue(mrFlagVariable);
```

Comment:
The usage of flag variables is outdated in favor of actual flags (like SLIP)

Reply:
Since there's no existing Kratos flag that corresponds to this use case I guess I should define a new local flag via `KRATOS_DEFINE_LOCAL_APPLICATION_FLAG` ?

Reply:
you could do that, but flags are a bit delicate in their usage (local and global flags that use the same bit will conflict with each other)

You could still use similar code, but I recommend a variable of type `bool` and saved in the non-historical data. This can be accesses with `GetValue`. The double in your current case is a leftover from when variables could not have other types

Reply:
Understood, thanks!